lntest/itest/lnd_route_blinding.go

```diff
@@ -680,6 +681,7 @@ func (b *blindedForwardTest) createBlindedRoute(hops []*forwardingEdge,
 			// TODO: We shouldn't have a next node ID here (but
 			// path ID isn't yet supported).
 			NextNodeID: dest,
+			PathID:     bytes.Repeat([]byte{1}, 32),
```

Comment:
You can remove next node ID here? I think that it's actually spec-noncomplient to have it in there for the last hop (validation _should_ complain?)

Reply:
Agreed that it can be removed. Re: whether it _must_ be removed - I have this for spec.

- If it is the final node:
    - MUST return an error if the payload contains other tlv fields than `encrypted_recipient_data`, `current_blinding_point`, `amt_to_forward`, `outgoing_cltv_value` and `total_amount_msat`. (DONE: covered by Carla’s allowed fields)
    - MUST return an error if `amt_to_forward`, `outgoing_cltv_value` or `total_amount_msat` are not present. (TODO: required field - calvin addition)
    - MUST return an error if `amt_to_forward` is below what it expects for the payment. (check already exists?)
    - MUST return an error if `incoming cltv_expiry` < `outgoing_cltv_value`. (check already exists?)
    - MUST return an error if `incoming cltv_expiry` < `current_block_height` + `min_final_cltv_expiry_delta`. (check already exists?)

Are you getting at that first bullet point? I read it as commenting **_only_** on the contents of the onion payload - not on it restricting the contents of the route blinding payload in any way.

Reply:
> Are you getting at that first bullet point? I read it as commenting only on the contents of the onion payload - not restricting the contents of the route blinding payload in any way.

I just put it in there to have _something_ in the last hop, so figured it could be deleted. If it passes validation then it's not a big deal either way.